src/command/colors.command.spec.ts

```diff
@@ -1,18 +1,8 @@
 import { execSync } from 'child_process';
-import * as colorHelper from '../helper/color.helper';
 import { colors as colorsCommand } from './colors.command';
 
-jest.mock('../helper/color.helper', () => ({
-  listColors: jest
-    .fn()
-    .mockReturnValue(
-      '\x1B[30mblack\x1B[0m, \x1B[31mred\x1B[0m, \x1B[32mgreen\x1B[0m, \x1B[33myellow\x1B[0m, \x1B[34mblue\x1B[0m, \x1B[35mmagenta\x1B[0m, \x1B[36mcyan\x1B[0m, \x1B[37mwhite\x1B[0m, \x1B[91mbrightRed\x1B[0m, \x1B[92mbrightGreen\x1B[0m, \x1B[93mbrightYellow\x1B[0m, \x1B[94mbrightBlue\x1B[0m, \x1B[95mbrightMagenta\x1B[0m, \x1B[96mbrightCyan\x1B[0m, \x1B[97mbrightWhite\x1B[0m\n',
-    ),
-}));
```

Comment:
We can consider adding a snapshots test about the checking diff of the color list.

https://jestjs.io/docs/snapshot-testing
https://www.daleseo.com/jest-snapshot/

Reply:
Thank you for letting me know😀
I added snapshot test to `color.helper.spec.ts`. 
```tsx
test('listColors should return a string with all colors', () => {
  const result = listColors();
  const colorKeys = Object.keys(COLOR)
    .filter((e) => e !== 'reset')
    .map((e) => `${COLOR[e]}${e}${COLOR.reset}`)
    .join(', ');

  expect(result).toMatch(colorKeys); // 1️⃣ 
  expect(result).toMatchSnapshot(); // 2️⃣ snapshot (check previous result on `./__snapshots__/color.helper.spec.ts.snap`)
});
```
![image](https://github.com/ts-animal/ts-animal/assets/94776135/791e5bb5-d53d-47b5-b1ba-8ee76971efab)

I have a question, 
When testing snapshots, is it preferable to test the colorKeys on 1️⃣, _only when a snapshot does not exist??_
Is it not redundant? to conduct both (1)snapshot and (2)functionality tests every time?🤔
I haven't tried writing snapshot test code before, so I'm not sure about it.

Reply:
> thank you for the comment✨

👉 I organized the test code for `src/helper/color.helper.ts` as follows:

(1) Verify if the output of the random color is in ANSI color format.
(2-1) Check if the color list is not null.
(2-2) Confirm if the displayed color list matches the previous snapshot. 👈

## update snapshot 😄

- make diff on COLORS

![image](https://github.com/ts-animal/ts-animal/assets/94776135/c2b1e5b2-f5be-4caa-93e5-995dd2052e9f)

- update!

![image](https://github.com/ts-animal/ts-animal/assets/94776135/780df748-929c-4330-b3e2-74f5c4ee12fb)
![image](https://github.com/ts-animal/ts-animal/assets/94776135/2ce0b99a-4bd4-4673-8b4c-7eb7f2ec759d)

## and also 

found the issue of a **type error** occurring each time key&value is added to the `COLOR` array,
I moved the `COLOR` constant to `src/type/color.type.ts` & change `ColorType`like below (and change COLOR to COLORS)

```tsx
export type ColorType = keyof typeof COLORS;
```
(If we need conventions for constants later, we can move constants to another location)